system/lib/libc/musl/arch/emscripten/bits/alltypes.h

```diff
@@ -114,7 +119,7 @@ typedef struct { union { int __i[12]; volatile int __vi[12]; void *__p[12]; } __
 #endif
 
 #if defined(__NEED_pthread_rwlock_t) && !defined(__DEFINED_pthread_rwlock_t)
-typedef struct { union { int __i[8]; volatile int __vi[8]; void *__p[8]; } __u; } pthread_rwlock_t;
+typedef struct { union { int __i[sizeof(long)==8?14:8]; volatile int __vi[sizeof(long)==8?14:8]; void *__p[sizeof(long)==8?7:8]; } __u; } pthread_rwlock_t;
```

Comment:
I wasn't aware sizeof can be used in the sizes of fields..?

But could this check if wasm64 using an ifdef?

Also, why is this 8 or 14? The difference of 6 seems surprising... (likewise abut 8 or 7)